src/animation/AnimationUtils.js

```diff
@@ -158,9 +158,63 @@ var AnimationUtils = {
 
 		}
 
+	},
+
+	clone: function ( source ) {
+
+		var cloneLookup = new Map();
+
+		var clone = source.clone();
+
+		parallelTraverse( source, clone, function ( sourceNode, clonedNode ) {
+
+			cloneLookup.set( sourceNode, clonedNode );
+
+		} );
+
+		source.traverse( function ( sourceMesh ) {
+
+			if ( ! sourceMesh.isSkinnedMesh ) return;
+
+			var sourceBones = sourceMesh.skeleton.bones;
+			var clonedMesh = cloneLookup.get( sourceMesh );
+
+			clonedMesh.skeleton = sourceMesh.skeleton.clone();
+
+			clonedMesh.skeleton.bones = sourceBones.map( function ( sourceBone ) {
+
+				var clonedBone = cloneLookup.get( sourceBone );
+
+				if ( ! clonedBone ) {
+
+					throw new Error( 'THREE.AnimationUtils: Required bones are not descendants of the given object.' );
+
+				}
```

Comment:
nit: Using `.has()` may look more natural?

    if ( ! cloneLookup.has( sourceBone ) ) {

        throw new Error( 'THREE.AnimationUtils: Required bones are not descendants of the given object.' );

    }

    var clonedBone = cloneLookup.get( sourceBone );

Reply:
Done ✅